src/Common/src/Interop/Linux/System.Net.Security.Native/Interop.HeimdalNtlm.cs

```diff
@@ -0,0 +1,173 @@
+                    incremental.AppendData(buffer, offset, count);
+                    hash = incremental.GetHashAndReset();
+                    _sequenceNumber++;
+                }
+
+                if (sealingKey == null)
+                {
+                    Array.Copy(hash, 0, output, ChecksumOffset, HMacDigestLength);
+                }
+                else
+                {
+                    byte[] cipher = sealingKey.SealOrUnseal(hash, 0, HMacDigestLength); 
+                    Array.Copy(cipher, 0, output, ChecksumOffset, cipher.Length);
+                }
+
+                return output;
+            }
+        }
+
+        internal sealed class SealingKey : IDisposable
+        {
+            private readonly byte[] _digest;
+            private SafeEvpCipherCtxHandle _cipherContext;
+
+            public SealingKey(byte[] key, string magicKey)
+            {
+                _digest = NtlmKeyDigest(key, magicKey);
+                _cipherContext = Interop.Crypto.EvpCipherCreate(Interop.Crypto.EvpRc4(), _digest, null, 1);
+            }
+
+            public byte[] SealOrUnseal(byte[] buffer, int offset, int count)
+            {
+                // Message Confidentiality. Reference: https://msdn.microsoft.com/en-us/library/cc236707.aspx
+                Debug.Assert(offset >= 0 && offset <= buffer.Length, "Cannot sign with invalid offset " + offset);
+                Debug.Assert(count >= 0, "cannot sign with invalid count");
+                Debug.Assert(count <= (buffer.Length - offset), "Cannot sign with invalid count ");
+
+                unsafe
+                {
+                    fixed (byte *bytePtr = buffer)
+                    {
+                         // Since RC4 is XOR-based, encrypt or decrypt is relative to input data
+                         // reference: https://msdn.microsoft.com/en-us/library/cc236707.aspx
+                          byte[] output = new byte[count];
+
+                          Interop.Crypto.EvpCipher(_cipherContext, output, (bytePtr + offset), count);
+                          MockUtils.MockLogging.PrintInfo(null, "returning from seal");
+                          return  output;
+                    }
+                }
+            }
+
+            public void Dispose()
+            {
+                if (_cipherContext != null)
+                {
+                    _cipherContext.Dispose();
+                    _cipherContext = null;
+                }
+            }
+        }
+
+        private static byte[] NtlmKeyDigest(byte[] key, string magicKey)
+        {
+            using (IncrementalHash incremental = IncrementalHash.CreateHash(HashAlgorithmName.MD5))
+            {
+                incremental.AppendData(key);
+                incremental.AppendData(Encoding.UTF8.GetBytes(magicKey));
+                return incremental.GetHashAndReset();
+            }
+        }
+
+        private static void MarshalUint(byte[] ptr, int offset, uint num)
```

Comment:
Nit: I realize this is private, but it'd be nice to follow the framework design guidelines here and use `MarshalUInt32` instead of `MarshalUint` (i.e. use type name for names of members, not language keywords), just for consistency in naming style.